pygfx/utils/gltf.py

```diff
@@ -41,22 +42,34 @@ def _parse_geometry(mesh):
 
 def _parse_material(pbrmaterial):
     material = gfx.MeshStandardMaterial()
-    material.map = _parse_texture(pbrmaterial.baseColorTexture)
 
-    material.emissive = gfx.Color(*pbrmaterial.emissiveFactor)
-    material.emissive_map = _parse_texture(pbrmaterial.emissiveTexture)
+    if pbrmaterial.baseColorTexture is not None:
+        material.map = _parse_texture(pbrmaterial.baseColorTexture)
 
-    metallic_roughness_map = _parse_texture(pbrmaterial.metallicRoughnessTexture)
-    material.roughness = pbrmaterial.roughnessFactor or 1.0
-    material.metalness = pbrmaterial.metallicFactor or 1.0
-    material.roughness_map = metallic_roughness_map
-    material.metalness_map = metallic_roughness_map
+    if pbrmaterial.emissiveFactor is not None:
+        material.emissive = gfx.Color(*pbrmaterial.emissiveFactor)
 
-    material.normal_map = _parse_texture(pbrmaterial.normalTexture)
-    material.normal_scale = (1, -1)
+    if pbrmaterial.emissiveTexture is not None:
+        material.emissive_map = _parse_texture(pbrmaterial.emissiveTexture)
 
-    material.ao_map = _parse_texture(pbrmaterial.occlusionTexture)
-    material.ao_map_intensity = 1.0
+    if pbrmaterial.metallicRoughnessTexture is not None:
+        metallic_roughness_map = _parse_texture(pbrmaterial.metallicRoughnessTexture)
+        material.roughness_map = metallic_roughness_map
+        material.metalness_map = metallic_roughness_map
+
+    if pbrmaterial.roughnessFactor is not None:
+        material.roughness = pbrmaterial.roughnessFactor
+
+    if pbrmaterial.metallicFactor is not None:
+        material.metalness = pbrmaterial.metallicFactor
+
+    if pbrmaterial.normalTexture is not None:
+        material.normal_map = _parse_texture(pbrmaterial.normalTexture)
+        # todo: why is this different from the default for MeshStandardMaterial?
+        material.normal_scale = (1, -1)
```

Comment:
My guess is differeny conventions or something. Maybe @panxinmiao has an idea?

Reply:
To be honest, I'm not sure it's right. I guess it may also be because of the inversion of the UV coordinate system.

When I was writing the helmet example, I used threejs to load the helmet model, and tracked the `normalScale` attribute, it is indeed set to (1, -1), so I set it to (1, -1) here too, but it seems that the final rendering effect is almost the same as the default value (1,1).

Reply:
It took me a bit of time to figure it out. We should probably use the default (1, 1).

The root cause here is still the problem of the UV coordinate system.

For the standard GLTF model, there may be `tangent` vertex attributes, corresponding to the normalMap provided by it. At this time, we usually do not care about the coordinate system, because they are consistent.

However, some models do not contain the `tangent` vertex attribute. At this time, we use the screen-space derivatives to automatically calculate it. threejs's uv coordinate system and GLTF specification are y-flipped, so they compensate it by setting `normalScale.y = - 1`. The wgpu does not have this problem.

In addition, at present, we have not resolved the vertex attribute `tangent`, and all tangents are automatically calculated.
Although the [Sponza scene](https://github.com/KhronosGroup/glTF-Sample-Models/tree/master/2.0/Sponza) example does have the `tangent` attribute.

I think it should be, but perhaps we need some more obvious examples to test to ensure that this is correct. 😅 

More details can be found here:
https://github.com/mrdoob/three.js/issues/11438
https://github.com/KhronosGroup/glTF-Sample-Models/pull/336